packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -613,6 +616,31 @@ export class RpcContractProvider
     return new BallotOperation(hash, operation, publicKeyHash, forgedBytes, opResponse, context);
   }
 
+  /**
+   *
+   * @description Submit a proposal to be voted on
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param BallotParams Proposals operation parameter
+   */
+  async proposals(params: ProposalsParams) {
+    const publicKeyHash = await this.signer.publicKeyHash();
+
+    if (params.source && validateAddress(params.source) !== ValidationResult.VALID) {
+      throw new InvalidAddressError(params.source);
+    }
+    const source = params.source ?? publicKeyHash;
+    const operation = await createProposalsOperation({
+      ...params,
+      source: source,
+    });
+    const prepared = await this.prepareOperation({ operation, source: publicKeyHash });
```

Comment:
Should source be source here rather than puclicKeyHash? Otherwise, we will override the source param if it was defined in the params.